### ISSUE TITLE
The Complete Roadmap to BlockChain Development 

### ISSUE THREAD

**Issue by KanizoRGB** [section-engineering-education/engineering-education#5056]
### Topic Suggestion
The Complete Roadmap to Blockchain Development 

### Writing sample(s): 
https://www.section.io/engineering-education/an-introduction-to-machine-learning-using-c++/
https://www.section.io/engineering-education/how-to-turn-your-photo-into-a-cartoon-using-python/

#### Proposed title of article
[Blockchain]The Complete Roadmap to Blockchain Development

#### Proposed article introduction
In the mainstream media, talks about cryptocurrencies and how people are becoming millionaires out of them, are ever buzzing. We often hear that cryptocurrencies will not only render the banks useless but also ensure anonymity and security in financial transactions. However, this is only the surface information. Cryptocurrencies are an example use case of an underlying technology called the blockchain.

Not so many people can explain what the blockchain is, let alone how the technology is applied in cryptocurrencies. I verily say unto you that the blockchain is not a fad based on the amount of money investors are putting into its research. In addition to that, new use cases emerge frequently the most recent being the application of blockchain to create Non Fungible Tokens(NFTs). Additional use cases of the blockchain include secure sharing of  medical data, anti-money laundering systems, music royalties tracking, etc

It is quite evident that blockchain technology will affect crucial aspects of our lives thus creating a myriad of opportunities for blockchain developers in the near future. In this tutorial, I will give you a road map on how you can become a blockchain developer and prepare yourself for the future. As the old adage says "Opportunity favors the prepared mind".

#### Key takeaways 

1. The reader will learn the needed programming languages to become a blockchain developer.
2. The reader will also become aware of some simple projects they can implement to test their understanding of the blockchain.
3. The reader will understand concepts applied in blockchain such as proof of work algorithm, what is a transaction, public and private keys, smart contracts, etc

#### Article quality
My article covers blockchain development as a whole thus imprinting a roadmap in the user's mind of what they need to learn in a systematic manner to be able to become blockchain developers. Other similar articles only cover small subsections thus making it hard for beginners to follow due to the knowledge gaps.

#### References
https://blockgeeks.com/guides/what-is-blockchain-technology/
https://www.forbes.com/sites/bernardmarr/2017/01/24/a-complete-beginners-guide-to-blockchain/?sh=3a1ab21a6e60

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by lalith1403** [comment]
Such roadmaps will be advantageous from a personal anecdotal narrative. Can you provide the journey you took and go in deep about the challenges you faced or were likely to face.

**Comment by KanizoRGB** [comment]
I took the approach of building a smart contract on top of the ethereum blockchain and integrating it with a crypto wallet. I started with some background knowledge about the blockchain and went ahead to guide the reader on the steps they would need to undertake in order to build the  smart contract and put it out there for use.
One of the challenges faced is that creating a smart contract requires a powerful laptop to  and a very fast internet connection, which if lacking would make the whole process of development cumbersome. Acquiring the two is also quite expensive.
In addition to that, some of the concepts applied in smart contracts are quite sophisticated and require a  great deal of time to research about and understand. The sophistication is compounded by the fact that there are not sufficient sources of information or reference on matters blockchain.
